keylime/cert_utils.py

```diff
@@ -71,46 +71,60 @@ def verify_ek(ekcert: bytes, tpm_cert_store: str) -> bool:
         return False
 
     try:
-        ek509 = x509_der_cert(ekcert)
         for cert_file, pem_cert in trusted_certs.items():
             signcert = x509_pem_cert(pem_cert)
-            if ek509.issuer != signcert.subject:
+            if cert.issuer != signcert.subject:
                 continue
 
             signcert_pubkey = signcert.public_key()
             try:
                 if isinstance(signcert_pubkey, RSAPublicKey):
-                    assert ek509.signature_hash_algorithm is not None
+                    assert cert.signature_hash_algorithm is not None
                     signcert_pubkey.verify(
-                        ek509.signature,
-                        ek509.tbs_certificate_bytes,
+                        cert.signature,
+                        cert.tbs_certificate_bytes,
                         padding.PKCS1v15(),
-                        ek509.signature_hash_algorithm,
+                        cert.signature_hash_algorithm,
                     )
                 elif isinstance(signcert_pubkey, EllipticCurvePublicKey):
-                    assert ek509.signature_hash_algorithm is not None
+                    assert cert.signature_hash_algorithm is not None
                     signcert_pubkey.verify(
-                        ek509.signature,
-                        ek509.tbs_certificate_bytes,
-                        ec.ECDSA(ek509.signature_hash_algorithm),
+                        cert.signature,
+                        cert.tbs_certificate_bytes,
+                        ec.ECDSA(cert.signature_hash_algorithm),
                     )
                 else:
                     logger.warning("Unsupported public key type: %s", type(signcert_pubkey))
                     continue
             except crypto_exceptions.InvalidSignature:
                 continue
 
-            logger.debug("EK cert matched cert: %s", cert_file)
+            logger.debug("Cert to verify matched cert: %s", cert_file)
             return True
     except Exception as e:
         # Log the exception so we don't lose the raw message
         logger.exception(e)
-        raise Exception("Error processing ek/ekcert. Does this TPM have a valid EK?").with_traceback(sys.exc_info()[2])
+        raise Exception("Error processing certificate.").with_traceback(sys.exc_info()[2])
 
-    logger.error("No Root CA matched EK Certificate")
+    logger.error("No Root CA matched Certificate")
```

Comment:
It may be good to pass in the type of certificate as a string parameter so that EK can re-appear here and above.